falcon.go

```diff
@@ -18,16 +18,25 @@
 // signature scheme.
 package falcon
 
-// #cgo CFLAGS: -O3
+// NOTE: cgo go code couldn't compile with the flags: -Wmissing-prototypes and -Wno-unused-paramete
+
+//#cgo CFLAGS:  -Wall -Wextra -Wpedantic -Wredundant-decls -Wshadow -Wvla -Wpointer-arith -Wno-unused-parameter -Wno-overlength-strings  -O3 -fomit-frame-pointer
 // #include "falcon.h"
 // #include "deterministic.h"
 import "C"
 
 import (
 	"fmt"
+	"runtime"
 	"unsafe"
 )
 
+const (
+	errKeygenFail  = "falcon keygen failed error is: %d"
+	errSignFail    = "falcon sign failed error is: %d"
+	errVerifyFail  = "falcon verify failed error is: %d"
+	errConvertFail = "falcon convert to CT failed error is: %d"
```

Comment:
I think you want to export these types, and use errors.New here, then wrap them with %d and %w below in the fmt.Errorf places

Reply:
do you mean to have something like this?
`ErrKeygenFail  = errors.New("falcon keygen failed")`
and
`return fmt.Errorf("%w error is: %d", errKeygenFail, int(r))`

Reply:
Yes or with the %w at the end like`fmt.Errorf("error code %d: %w"), int(r), ErrKeygenFail)`